modules/signals/entities/src/with-entities.ts

```diff
@@ -0,0 +1,67 @@
+import { computed } from '@angular/core';
+import {
+  SignalStoreFeature,
+  signalStoreFeature,
+  withComputed,
+  withState,
+} from '@ngrx/signals';
+import {
+  EntityId,
+  EntityMap,
+  EntitySignals,
+  EntityState,
+  NamedEntitySignals,
+  NamedEntityState,
+} from './models';
+import { getEntityStateKeys } from './helpers';
+
+export function withEntities<Entity>(): SignalStoreFeature<
+  { state: {}; signals: {}; methods: {} },
+  {
+    state: EntityState<Entity>;
+    signals: EntitySignals<Entity>;
+    methods: {};
+  }
+>;
+export function withEntities<Entity, Collection extends string>(config: {
+  entity: Entity;
+  collection: Collection;
+}): SignalStoreFeature<
+  { state: {}; signals: {}; methods: {} },
+  {
+    state: NamedEntityState<Entity, Collection>;
+    signals: NamedEntitySignals<Entity, Collection>;
+    methods: {};
+  }
+>;
+export function withEntities<Entity>(config: {
+  entity: Entity;
+}): SignalStoreFeature<
+  { state: {}; signals: {}; methods: {} },
+  {
+    state: EntityState<Entity>;
+    signals: EntitySignals<Entity>;
+    methods: {};
+  }
+>;
+export function withEntities<Entity>(config?: {
+  entity: Entity;
+  collection?: string;
+}): SignalStoreFeature {
+  const { entityMapKey, idsKey, entitiesKey } = getEntityStateKeys(config);
+
+  return signalStoreFeature(
+    withState({
+      [entityMapKey]: {},
+      [idsKey]: [],
+    } as any),
+    withComputed((store) => ({
+      [entitiesKey]: computed(() => {
+        const entityMap = store[entityMapKey]() as EntityMap<Entity>;
```

Comment:
There won't be duplicated keys because `signalStore` works as a spread operator - it overrides previously defined properties:

```ts
const Store = signalStore({
  withEntities<User>(),
  withEntities<Todo>()
});

const store = inject(Store);

store.entityMap() // type: EntityMap<Todo>
```

The same is true for named entity collections:

```ts
const Store = signalStore({
  withEntities({ entity: type<User>(), collection: 'todo' }),
  withEntities({ entity: type<Todo>(), collection: 'todo' })
});

const store = inject(Store);

store.todoEntityMap() // type: EntityMap<Todo>
```

Reply:
The comment I made was exactly because it overrides.
I see peoples copy pasting `withEntities` without changing the collection, which could cause some issues because the one will override the other.

Reply:
Oh got it. I'd keep it as it is for now. We can introduce a warning in development mode if we realize that this is a common case.